wizard/src/components/preview_section.tsx

```diff
@@ -85,19 +106,38 @@ export function PreviewSection(props: PreviewSectionProps): JSX.Element {
       type: "text/json",
     });
     files["import.tmcf"] = generateTMCF(props.correctedMapping);
-    Object.entries(files).forEach(([fileName, fileContent]) => {
-      zipFolder.file(fileName, fileContent);
-    });
-    zipFolder.generateAsync({ type: "blob" }).then((content) => {
-      const link = document.createElement("a");
-      const url = window.URL.createObjectURL(content);
-      link.setAttribute("href", url);
-      link.setAttribute("download", "importPackage.zip");
-      link.onclick = () => {
-        setTimeout(() => window.URL.revokeObjectURL(url));
-      };
-      link.click();
-      link.remove();
-    });
+    cleanCsvPromise
+      .then((csvString) => {
+        if (props.shouldGenerateCsv) {
+          files["cleanedData.csv"] = new Blob([csvString], {
+            type: "text/csv;chartset=utf-8",
+          });
+        }
+        Object.entries(files).forEach(([fileName, fileContent]) => {
+          zipFolder.file(fileName, fileContent);
+        });
+        zipFolder
+          .generateAsync({ type: "blob" })
+          .then((content) => {
+            const link = document.createElement("a");
+            const url = window.URL.createObjectURL(content);
+            link.setAttribute("href", url);
+            link.setAttribute("download", "importPackage.zip");
+            setIsGeneratingFiles(false);
+            link.onclick = () => {
+              setTimeout(() => window.URL.revokeObjectURL(url));
+            };
+            link.click();
+            link.remove();
+          })
+          .catch(() => {
+            setIsGeneratingFiles(false);
+            alert("Sorry, there was a problem generating the import package.");
+          });
```

Comment:
For my understanding, since this is wrapped with outside catch(), can we just let the exception here propagate up and skip this?